app/src/main/java/net/squanchy/eventdetails/domain/view/ExperienceLevel.java

```diff
@@ -1,29 +1,38 @@
 package net.squanchy.eventdetails.domain.view;
 
-import android.support.annotation.IntRange;
 import android.support.annotation.StringRes;
 
 import net.squanchy.R;
 
 public enum ExperienceLevel {
-    BEGINNER(0, R.string.experience_level_beginner),
-    INTERMEDIATE(1, R.string.experience_level_intermediate),
-    ADVANCED(2, R.string.experience_level_advanced);
+    BEGINNER("beginner", R.string.experience_level_beginner),
+    INTERMEDIATE("intermediate", R.string.experience_level_intermediate),
+    ADVANCED("advanced", R.string.experience_level_advanced);
+
+    public static ExperienceLevel fromRawLevel(String rawLevel) {
+        switch (rawLevel.toLowerCase()) {
+            case "beginner":
+                return ExperienceLevel.BEGINNER;
+            case "intermediate":
+                return ExperienceLevel.INTERMEDIATE;
+            case "advanced":
+                return ExperienceLevel.ADVANCED;
+            default:
+                throw new IllegalArgumentException("Invalid raw level description: " + rawLevel);
+        }
+    }
 
-    private final int rawLevel;
+    private final String rawLevel;
     private final int labelStringResId;
 
-    public static ExperienceLevel fromRawLevel(@IntRange(from = 0, to = 2) int rawLevel) {
-        return ExperienceLevel.values()[rawLevel];
-    }
+    ExperienceLevel(String rawLevel,
```

Comment:
Actually, AS tells me that `private` is redundant for enum constructors... I would leave it as it is, as it looks already `private`?

<img width="442" alt="screen shot 2017-03-02 at 09 03 18" src="https://cloud.githubusercontent.com/assets/2534841/23498257/25a63680-ff27-11e6-86a6-41a7a19950af.png">